api/src/opentrons/protocol_engine/types.py

```diff
@@ -337,13 +338,26 @@ class LoadedLabware(BaseModel):
     )
 
 
-class Liquid(BaseModel):
+class HexColor(BaseModel):
+    """Hex color representation."""
+
+    displayColor: Optional[str] = Field(default_factory=None)
+
+    @validator("displayColor")
+    def _color_is_a_valid_hex(cls, v: Optional[str]) -> Optional[str]:
+        if v:
+            match = re.search(r"^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$", v)
```

Comment:
```suggestion
            match = re.match(r"^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$", v)
```

We can use `re.match()` to be slightly more specific than `re.search()` here, because we know the pattern will begin at the beginning of the string, not somewhere in the middle.